test/retrieval-stats.test.js

```diff
@@ -66,15 +66,15 @@ describe('retrieval statistics', () => {
 
     assertPointFieldValue(point, 'ttfb_min', '1000i')
     assertPointFieldValue(point, 'ttfb_mean', '4000i')
-    assertPointFieldValue(point, 'ttfb_p90', '10000i')
+    assertPointFieldValue(point, 'ttfb_p90', '8199i')
 
     assertPointFieldValue(point, 'duration_p10', '2000i')
     assertPointFieldValue(point, 'duration_mean', '18500i')
-    assertPointFieldValue(point, 'duration_p90', '50000i')
+    assertPointFieldValue(point, 'duration_p90', '41000i')
 
-    assertPointFieldValue(point, 'car_size_p10', '1024i')
+    assertPointFieldValue(point, 'car_size_p10', '1331i')
     assertPointFieldValue(point, 'car_size_mean', '52430080i')
-    assertPointFieldValue(point, 'car_size_p90', '209715200i')
+    assertPointFieldValue(point, 'car_size_p90', '146801254i')
```

Comment:
in comparison to the original expensive approach, does this make our valuers more right or wrong now?

Reply:
I think the new approach makes our values more right.

```
> getValueAtPercentile([10, 20, 30], 90)
28
```

Reply:
Before, we would pick `20` or `30` as the value at p90.

The downside is that 28 picked by the new algorithm was not reported anywhere in the data, it's a synthetic value 🤷🏻‍♂️

Reply:
Verified using https://www.calculatorsoup.com/calculators/statistics/percentile-calculator.php

<img width="361" alt="Screenshot 2023-12-07 at 14 54 41" src="https://github.com/filecoin-station/spark-evaluate/assets/1140553/9f33c178-de79-441e-9d77-20d3f7fbebab">

Reply:
OTOH:

https://goodcalculators.com/percentile-calculator/

<img width="851" alt="Screenshot 2023-12-07 at 14 55 26" src="https://github.com/filecoin-station/spark-evaluate/assets/1140553/adbfbf62-9d7e-432f-b2c5-2fa6f22a2599">

Reply:
Postgres offers two functions:

`percentile_cont`
Computes the _continuous percentile_, a value corresponding to the specified _fraction_ within the ordered set of aggregated argument values. This will interpolate between adjacent input items if needed.

`percentile_disc`
Computes the _discrete percentile_, the first value within the ordered set of aggregated argument values whose position in the ordering equals or exceeds the specified _fraction_. The aggregated argument must be of a sortable type.

I think it's up to us to decide whether we want to use _continuous_ or _discrete_ percentiles.

@patrickwoodhead do you have any thoughts on this?

Reply:
Given we likely have more than 100 values, do we need to worry about interpolation? Let's say we have the numbers 1 to 200 and we want the 1st percentile. We first get the number 1 and 2. IIUC the above definitions, the continuous approach then gives us 1.5. the discrete approach gives us 1.

Suppose our data set is instead 1,10,11,20,21,30,31,.... for 200 values. The the two 1st percentile outputs respectively are 5.5 and 1. I think 5.5 is a better answer here.

So unless the dataset is really not a good fit for continuous, I vote we go for continuous, even if it means we aren't returning a data point in the data set.